x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java

```diff
@@ -821,6 +822,14 @@ private static void checkFullTextQueryFunctions(LogicalPlan plan, Set<Failure> f
                 m -> "[" + m.functionName() + "] " + m.functionType(),
                 failures
             );
+            checkCommandsBeforeExpression(
+                plan,
+                condition,
+                Term.class,
+                lp -> (lp instanceof Limit == false) && (lp instanceof Aggregate == false),
```

Comment:
Just passing by... I think this needs `&& (lp instanceof MvExpand == false)` as well (and also the above, most likely).
The above was supposed to be a [temporary solution](https://github.com/elastic/elasticsearch/pull/116642#discussion_r1841662017), so maybe we can centralize it in a separate method with a TODO.
In general, it would be better to have a whitelist rather than a blacklist here.

Reply:
@luigidellaquila Agree with the refactoring suggestions here. Given that this affects match also, let's do it in a separate PR with more targeted testing.

Reply:
makes sense to me to do it as a follow up - and when we do - it should be an allow list so we don't have to always catch up of adding here a check for every new command we add - related https://github.com/elastic/elasticsearch/pull/116642#discussion_r1841055613

Reply:
I linked this comment to #115311 so that it is not lost.